utils/schemas.py

```diff
@@ -0,0 +1,406 @@
+        description="State or province of the person or organization that distributes the data.",
+        default=None,
+    )
+    platform: str = Field(
+        description="Name of the platform(s) that supported the sensor data used to create this data set or product. Platforms can be of any type, including satellite, ship, station, aircraft or other. Indicate controlled vocabulary used in platform_vocabulary.",
+    )
+    platform_vocabulary: str = Field(
+        description="Controlled vocabulary for the names used in the 'platform' attribute.",
+    )
+    platform_id: str = Field(
+        title="Platform ID",
+        description="An optional, short identifier for the platform, if the data provider prefers to define an id that differs from the dataset identifier, as specified by the id attribute. Platform_id should be a single alphanumeric string with no blank spaces.",
+        default=None,
+    )
+    platform_name: str = Field(
+        title="Platform Name",
+        description="A descriptive, long name for the platform used in collecting the data. The value of platform_name will be used to label the platform in downstream applications, such as IOOS’ National Products (Environmental Sensor Map, EDS, etc).",
+    )
+    WMO_platform_code: str = Field(
+        title="WMO Platform Code",
+        description="The WMO identifier for the platform used to measure the data. This identifier can be any of the following types: 1. WMO ID for buoys (numeric, 5 digits), 2. WMO ID for gliders (numeric, 7 digits), 3. NWS ID (alphanumeric, 5 digits). When a dataset is assigned a wmo_platform_code it is thereby assigned a secondary Asset Identifier for the 'WMO' naming_authority. See https://ioos.github.io/ioos-metadata/ioos-metadata-profile-v1-2.html#rules-for-ioos-asset-identifier-generation for more details.",
+        default=None,
+    )
+    
+
+class ACDDDatasetConfig(DatasetConfig, extra=Extra.allow):
+    attrs: ACDDGlobalAttrs = Field(
+        ...,
+        description="Attributes that pertain to the dataset as a whole (as opposed to"
+        " attributes that are specific to individual variables.",
+    )
+    
+class IOOSDatasetConfig(DatasetConfig, extra=Extra.allow):
+    attrs: IOOSGlobalAttrs = Field(
+        
+        description="Attributes that pertain to the dataset as a whole (as opposed to"
+        " attributes that are specific to individual variables.",
+    )
+
+
+def generate_schema(dataset_config: DatasetConfig, dir: Path = Path(".vscode/schema/")):
+    dir.mkdir(exist_ok=True)
+    cls_mapping = {
+        "retriever": RetrieverConfig,
+        "dataset": dataset_config,
+        "quality": QualityConfig,
+        "storage": StorageConfig,
+        "pipeline": PipelineConfig,
+    }
+    for key, config_class in cls_mapping.items():
+        path = dir / f"{key}-schema.json"
+        config_class.generate_schema(path)
+        print(f"Wrote {key} schema file to {path}")
+    print("Done!")
```

Comment:
Let's turn this into a `typer` CLI to make it easier to use directly. We can use an Enum type to map a new `standards` parameter to what we want to use for the dataset schema. e.g., `default-->tsdat.DatasetConfig`, `acdd-->ACDDDatasetConfig`, or `ioos-->IOOSDatasetConfig`.

See the [schema generator in `tsdat/utils.py`](https://github.com/tsdat/tsdat/blob/29f4cd8aeb89b54faf5dd37abad85c4faf2f8e5c/tsdat/utils.py#L324-L371) for an example. We'll make a change here to do `standards` instead of the `schema_type` in the tsdat example

Reply:
Also, above the `generate_schema()` function definition we'll want to add the following lines:

```python
import typer

app = typer.Typer()

...

class StandardsType(str, Enum):
    basic = "basic"
    acdd = "acdd"
    ioos = "ioos"

@app.command()
def generate_schema(
    dir: Path = typer.Option(Path(".vscode/schema/"), file_okay=False, dir_okay=True),
    standards: StandardsType = typer.Option(StandardsType.basic),
):
    ...


```